tests/chainer_tests/links_tests/loss_tests/test_crf1d.py

```diff
@@ -0,0 +1,127 @@
+import unittest
+
+import itertools
+import numpy
+
+import chainer
+from chainer.backends import cuda
+from chainer import initializers
+from chainer import links
+from chainer import testing
+from chainer.testing import attr
+from chainer.testing import condition
+
+
+@testing.parameterize(*testing.product({
+    'dtype': [numpy.float16, numpy.float32, numpy.float64],
+    'initial_cost': ['random', None],
+}))
+class TestCRF1d(unittest.TestCase):
+    def _calc_score(self, batch, ys):
+        cost = self.link.cost.data
```

Comment:
Could you prefer to use the newer `Variable.array` attribute, for consistency?
```suggestion
        cost = self.link.cost.array
```

Reply:
Oh sorry, applied.